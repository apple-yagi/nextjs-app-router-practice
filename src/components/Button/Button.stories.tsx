import { Meta, StoryObj } from "@storybook/react";

import { Button } from ".";

const meta = {
  component: Button,
} satisfies Meta<typeof Button>;

export default meta;

export const Default = {
  args: {
    children: "三毛猫",
  },
} satisfies StoryObj<typeof meta>;

export const Primary = {
  args: {
    variants: {
      color: "primary",
    },
    children: "マンチカン",
  },
} satisfies StoryObj<typeof meta>;

export const Secondary = {
  args: {
    variants: {
      color: "secondary",
    },
    children: "スコティッシュフォールド",
  },
} satisfies StoryObj<typeof meta>;

export const Danger = {
  args: {
    variants: {
      color: "danger",
    },
    children: "ベンガル",
  },
} satisfies StoryObj<typeof meta>;
