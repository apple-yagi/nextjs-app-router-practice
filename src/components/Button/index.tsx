import { ComponentPropsWithoutRef } from "react";
import { tv, type VariantProps } from "tailwind-variants";

const button = tv({
  base: "flex items-center justify-center rounded px-4 py-2 text-base",
  variants: {
    color: {
      primary:
        "border border-blue-500 bg-blue-500 text-white hover:border-blue-700 hover:bg-blue-700",
      secondary: "border border-gray-500 bg-white text-black hover:bg-gray-300",
      danger:
        "border border-red-500 bg-red-500 text-white hover:border-red-700 hover:bg-red-700",
    },
  },
  defaultVariants: {
    color: "primary",
  },
});

type ButtonVariants = VariantProps<typeof button>;

export const Button = ({
  children,
  role = "button",
  variants,
  ...props
}: ComponentPropsWithoutRef<"button"> & { variants?: ButtonVariants }) => {
  return (
    <button className={button(variants)} role={role} {...props}>
      {children}
    </button>
  );
};
