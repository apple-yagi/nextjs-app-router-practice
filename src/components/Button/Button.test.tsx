import { composeStories } from "@storybook/react";
import { render } from "@testing-library/react";

import * as stories from "./Button.stories";

const { Default, Primary, Secondary, Danger } = composeStories(stories);

describe("Button", () => {
  it("Default Buttonが正しく表示されること", () => {
    expect(render(<Default />).asFragment()).toMatchSnapshot();
  });

  it("Primary Buttonが正しく表示されること", () => {
    expect(render(<Primary />).asFragment()).toMatchSnapshot();
  });

  it("Secondary Buttonが正しく表示されること", () => {
    expect(render(<Secondary />).asFragment()).toMatchSnapshot();
  });

  it("Danger Buttonが正しく表示されること", () => {
    expect(render(<Danger />).asFragment()).toMatchSnapshot();
  });
});
