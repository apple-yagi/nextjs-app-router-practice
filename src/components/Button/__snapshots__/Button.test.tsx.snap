// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > Danger Buttonが正しく表示されること 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center rounded px-4 py-2 text-base disabled:opacity-50 border border-red-500 bg-red-500 text-white hover:border-red-700 hover:bg-red-700"
    role="button"
  >
    ベンガル
  </button>
</DocumentFragment>
`;

exports[`Button > Default Buttonが正しく表示されること 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center rounded px-4 py-2 text-base disabled:opacity-50 border border-blue-500 bg-blue-500 text-white hover:border-blue-700 hover:bg-blue-700"
    role="button"
  >
    三毛猫
  </button>
</DocumentFragment>
`;

exports[`Button > Primary Buttonが正しく表示されること 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center rounded px-4 py-2 text-base disabled:opacity-50 border border-blue-500 bg-blue-500 text-white hover:border-blue-700 hover:bg-blue-700"
    role="button"
  >
    マンチカン
  </button>
</DocumentFragment>
`;

exports[`Button > Secondary Buttonが正しく表示されること 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center rounded px-4 py-2 text-base disabled:opacity-50 border border-gray-500 bg-white hover:bg-gray-300"
    role="button"
  >
    スコティッシュフォールド
  </button>
</DocumentFragment>
`;
